### ISSUE TITLE
The flatpak version does not create the dosbox-staging.conf automatically if it starts from outside the Linux Terminal

### ISSUE THREAD

**Issue by ThomasEricB** [dosbox-staging/dosbox-staging#2417]
### Are you using the latest Dosbox-Staging Version?

- [X] I have checked releases and am using the latest release.

### What Operating System are you using?

Linux x86_64

### If Other OS, please describe

SteamOS 3.0

### Relevant hardware info

Steam Deck: Processor: AMD APU - CPU: Zen 2 4c/8t, 2.4-3.5GHz (up to 448 GFlops FP32) - GPU: 8 RDNA 2 CUs, 1.0-1.6GHz (up to 1.6 TFlops FP32) - RAM: 16 GB LPDDR5

### Have you checked that no other similar issue already exists?

- [X] I have searched and not found similar issues.

### A clear and concise description of what the bug is.

Weird stuff is afoot with this one. DOSbox Staging's Flatpak should write a .conf file automatically on `/home/deck/.var/app/io.github.dosbox-staging/config/dosbox/dosbox-staging.conf`, however it isn't doing so. We found out by @johnnovak 's request for the website. Reproduced by me and by @dreamer . 

### Steps to reproduce the behaviour.

Explain how to reproduce
1. Install Flatpak
2. Launch DOSBox Staging from either the Discover Store or Start Menu (NOT from the terminal)
3. Close DOSBox Staging
4. No .conf inside the `/home/deck/.var/app/io.github.dosbox-staging/config/dosbox/` directory.

### Your configuration

```ini
N/A
```

### Provide a Log

```text
N/A
```

### Code of Conduct & Contributing Guidelines

- [X] Yes, I agree.

**Comment by kcgen** [comment]
For repro step 2, can you post the resulting log lines prefixed with CONFIG: ` ?

**Comment by rderooy** [comment]
```bash
$ flatpak run io.github.dosbox-staging
date       time         | 
2023-04-17 21:44:30.234 | arguments: dosbox
2023-04-17 21:44:30.234 | Current dir: /home/rderooy
2023-04-17 21:44:30.234 | stderr verbosity: 0
2023-04-17 21:44:30.234 | -----------------------------------
2023-04-17 21:44:30.234 | dosbox-staging version 0.80.1
2023-04-17 21:44:30.234 | ---
2023-04-17 21:44:30.234 | LOG: Loguru version 2.1.0 initialized
2023-04-17 21:44:30.258 | SDL: version 2.26.5 initialized (wayland video and pulseaudio audio)
2023-04-17 21:44:30.258 | LANG: Using internal English language messages
2023-04-17 21:44:30.259 | CONFIG: Wrote new primary conf file '/home/rderooy/.var/app/io.github.dosbox-staging/config/dosbox/dosbox-staging.conf'
2023-04-17 21:44:30.259 | CONFIG: Loaded new primary conf file /home/rderooy/.var/app/io.github.dosbox-staging/config/dosbox/dosbox-staging.conf
```
```bash
$ file /home/rderooy/.var/app/io.github.dosbox-staging/config/dosbox/dosbox-staging.conf
/home/rderooy/.var/app/io.github.dosbox-staging/config/dosbox/dosbox-staging.conf: ASCII text
```

Note that it does not create the file, if it finds an existing `dosbox.conf` or `dosbox-staging.conf` file in `~/.dosbox` or `~/.config/dosbox`

**Comment by rderooy** [comment]
> I'm assuming `$HOME/.var/app/io.github.dosbox-staging/config` is being provided via XDG environment variables from flatpak?

Correct.

**Comment by ThomasEricB** [comment]
@rderooy @dreamer @kcgen , on the Steam Deck we don't run with `$ flatpak run io.github.dosbox-staging`, instead we launch through the Discover Store or from the Start Menu. Either does not spawn the linux terminal with the logs, which might be part of the reason why it's not generating the .conf file on `~/.var/app/io.github.dosbox-staging/config/dosbox/dosbox-staging.conf`. 

![image](https://user-images.githubusercontent.com/9662095/232598298-d972ba7e-6c28-4ce0-9d27-dd2150459619.png)

HOWEVER, as like a PLOT TWIST from Game of Thrones running  `$ flatpak run io.github.dosbox-staging`, generated the following fascinating line on the terminal:  `2023-04-17 17:18:09.788 | CONFIG: Loaded primary conf file /home/deck/.dosbox/dosbox-staging.conf`, but as you can see below THE .dosbox directory does not exist on my system at this time (!!!):

![image](https://user-images.githubusercontent.com/9662095/232601354-3f047221-fa48-4b49-a58f-92ad565e032e.png)

**Damn it, we are in times of phantom directories and ghost configs!** I've then uninstalled the damned flatpak thing and deleted all the files:

![image](https://user-images.githubusercontent.com/9662095/232603177-3732696d-4128-4d8c-bfc7-af053215ec8d.png)

However, even on a fresh install **IT DID NOT** even created a dosbox folder inside 
`/home/deck/.var/app/io.github.dosbox-staging/config`, as it should:

![image](https://user-images.githubusercontent.com/9662095/232604587-85fbc1ff-a3f1-4d01-bdff-de4af01100b3.png)

Ok, so I try the method @rderooy mentioned one last time, by running `$ flatpak run io.github.dosbox-staging` on the terminal. Crazy enough it is STILL finding the .dosbox that DOESN'T exist. WTF. 

![image](https://user-images.githubusercontent.com/9662095/232606467-6a0d49e0-e600-4a9b-9517-07175122522e.png)

IN SHORT, I don't have a clue about what's going on.

**Comment by kcgen** [comment]
@ThomasEricB , can you add an argument to however the Discovery Store is launching Staging?

Add `-editconf`, and it will try to open it in the `nano` editor (or the editor set in the  EDITOR= environment variable). 

If it's saying it's opening a primary conf file, then it's finding it. 

The question is: how many sub-filesystems are created or hidden from your external or top level environment?

Another option is to replace the dosbox staging executable with `bash` and pop open a shell within this sub-enviroment. Then you can inspect around. 

(This kind of stuff is exactly why the project needs external people supporting it.. it's a rabbit hole!)

**Comment by ThomasEricB** [comment]
@dreamer Continuation from my last comment. **MANAGED to pinpoint the issue, so keep reading.** Running with `CONFIG -WC` in the `$ flatpak run io.github.dosbox-staging` populated and created the `~/.dosbox` directory with the folders and the .conf inside it. Deleting that directory and running DOSBox Staging again from the terminal, it did the same thing again. 

Then deleting it a 2nd time and **running the flatpak from the Discover Store**, while still DID NOT create the config file automatically, running  `CONFIG -WC` generated it at `/home/deck/.var/app/io.github.dosbox-staging/config/dosbox/dosbox-staging.conf`, which is the correct path:

![image](https://user-images.githubusercontent.com/9662095/232614852-a21f6b75-cb2c-43d7-b184-dcfb3c0fee83.png)

After that I deleted the dosbox folder inside the Flatpak files and I ran `$ flatpak run io.github.dosbox-staging` on the terminal and Voilà! It now created the .conf file with the behaviour as described by @rderooy above:

![image](https://user-images.githubusercontent.com/9662095/232616019-e41edc8f-08c9-4d5f-a4ea-4102c6308a64.png)

TL;DR: **SO, this means that the present issue, where it doesn't automatically generate the .conf, is being caused by running the flatpak from the Discover Store/ Start Menu!**

**Comment by kcgen** [comment]
I wonder what Discovery Store is doing? (perhaps it's a black box - maybe even using a cloud virtual filesystem or something?)

At least with your known solution above, it can be documented and simply a one-time step for deck users.

**Comment by Burrito78** [comment]
Steamdeck support is coming anyway, afaik. Problems will be ironed out by then. Closing for now, but thanks for the Feedback! :)

**Comment by ThomasEricB** [comment]
Test

**Comment by kcgen** [comment]
By changing the XDG environment paths, FlatPak uses the XDG standard to guide compliant applications into its containerized areas.

@rderooy's results show DOSBox Staging is correctly adhering to the XDG standard by both creating in and reading its primary conf from the XDG-configured area, shown in @rderooy's results as: `$HOME/.var/app/io.github.dosbox-staging/config/...`

@ThomasEricB , your results isolating the problem to how FlatPak is being launched from the Discover Store's start menu means that (probably) some interaction there is not setting up the XDG environment properly.

Suggest asking about it here: https://github.com/flathub/io.github.dosbox-staging.

However I would avoid declaring it bug against FlatPak because your results showed works fine when launched from the command line_, meaning that the real culprit is the Discovery Store not properly setting up the environment.
